### ISSUE TITLE
Simon The Sorcerer: Strange HDF size

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#674]
I've been trying to attach Simon The Sorceror (1994)(Adventure Soft).hdf (PlanetEmu, HDF section) in vAmiga without success.

The HDF is a DOS0 disk (OFS), has no RDB, and the file size is 12000000.

Observations: 

- 12000000 is not dividable by 512.
- UAE assigns a disk geometry of 732 / 1 / 32, but 732 * 32 * 512 is less than 12000000.

Conclusion: 

 - There is some real magic going on here.

**Comment by dirkwhoffmann** [comment]
More findings: After hard-coding the drive geometry, I can load the game. So the quest is to find some magic function that maps 12000000 to (732,1,32).

Unfortunately, the game itself doesn't look as expected 😕:

<img width="1034" alt="Bildschirmfoto 2022-03-31 um 17 32 52" src="https://user-images.githubusercontent.com/12561945/161096399-1e55645e-ffde-4401-977e-6defc4d68ee1.png">

Maybe it's the AGA version? I tried to double-check in UAE, but I am not smart enough to mount an HDF there. The hard drive does not appear on the workbench screen. There are tons of controllers to select from and another myriad of checkboxes to enable or disable. I could try all combinations, but that would take another myriad of years.

**Comment by emoon** [comment]
You can likely check if it's the AGA version if it tries to enable 8 bitplanes or so.

**Comment by dirkwhoffmann** [comment]
> As for testing in WinUAE

Got it to work now. I had to use the `uaehf device`. It was set to IDE before which didn't work.

<img width="594" alt="Bildschirmfoto 2022-04-01 um 09 23 56" src="https://user-images.githubusercontent.com/12561945/161218143-d4a83100-ce90-4ce1-8a4a-00917dead30b.png">

**Comment by dirkwhoffmann** [comment]
> You could probably do that, and check if you get a root block that way, and only if that doesn't work fall back to more complicated logic?

Yes, my new code does it this way. It first seeks the root block. After that, it tries to predict the number of blocks via different heuristics and verifies the outcome by comparing the predicted location of the root block with the actual location.

```
   auto match = [&]() {
        return (numReserved + highKey) / 2 == rootKey;
    };

   if (auto root = seekRB(); root) {
        
        rootKey = isize(root - data.ptr) / bsize();
        
        // Predict block count by analyzing the file size
        highKey = data.size / bsize() - 1;
        if (match()) return highKey + 1;
        
        // Predict block count by assuming a 32 sector standard geometry
        highKey = 32 * (data.size / (32 * bsize())) - 1;
        if (match()) return highKey + 1;

        // Predict by faking the numbers to fit
        highKey = 2 * rootKey - numReserved;
        if (match()) return highKey + 1;

        fatalError;
    }
```

Regarding the graphics issue, I've found out that it does work with Kickstart 1.3 in vAmiga, but doesn't with Kickstart 2.04. I'll open up another issue for that...

<img width="1034" alt="Bildschirmfoto 2022-04-01 um 09 05 23" src="https://user-images.githubusercontent.com/12561945/161218848-81ee8e3b-0bcc-4d9c-877f-c0e8b08629ab.png">

**Comment by dirkwhoffmann** [comment]
Fixed in v2.0b2 which is now online.
